API.md

```diff
@@ -368,6 +368,33 @@ If you use data functions rather than passing data on the `request` object, thes
     `request.app['hapi-govuk-question-page'].config`
     
     This parameter is not required if the page does not have any components that need request-specific configuration.
+
+### `error validation result` object
+
+If the setData validation fails, an object may be returned to display details of the error on the form error fields.
+The object should an `errors` property with two child properties:
+- `titleText`: the headline error message for the main error box.
+- `errorList`: and array containing `errorDetails` of the errors encountered.
+
+The `errorDetails` must contain the following three fields:
+- `href`: the ID of the item in the form, i.e. `#dateOfBirth`.
+- `name`: the name of the form field, i.e. `dateOfBirth`.
+- `text`: the error message, i.e. `date of birth must be in the past`.
+
+An example is shown below:
+```
+{ errors: 
+  { titleText: 'Fix the following errors', 
+    errorList: [
+      { 
+        href: '#dateOfBirth', 
+        name: 'dateOfBirth', 
+        text: 'date of birth must be in the past'
+      }
+    ]
+  }
+}
+```
```

Comment:
I don't think this is correct as the plugin's client application won't know how the plugin is handling the display and naming of form elements and errors.

The plugin code should be taking the Joi validation result returned from this function and, if it's an error, converting it into an appropriate format to be used in the view.

If the code isn't doing that then it needs to be fixed so that it does.